### DISCUSSION TITLE
Add video option

### DISCUSSION THREAD

**Discussion by davi19** [outline/outline#2539] [Ideas]
Currently in the system there is no option to add a video. As in the case of images and gifs. It would be possible to have this option as it has on the nuclin. Adding the video the system understands and adds a player for that video. As can be seen in the image below.

![Captura de tela de 2021-09-08 09-59-34](https://user-images.githubusercontent.com/9946675/132531338-3b0d4668-59e2-456a-b370-2b1251210eef.png)

**Comment by tommoor** [comment]
Definitely, this will come. I wonder if any sort of format conversion is needed or we can just serve the uploaded files directly back to the user

**Reply by davi19** [reply]
Well you can use something like the HTML5 video tag. Upload the video and create a video tag referencing the file link.

**Reply by davi19** [reply]
The Get the format in upload by extension of file and in construction of element video add the format. Maybe initially limited the formats accepts

**Comment by nvolker** [comment]
Would absolutely love video support here @tommoor! We're starting to create more and more short-form video guides and onboarding content, so having it live inside our company Handbook and Technical Hub would be incredibly valuable!

**Reply by tommoor** [reply]
Starting work on #859 this week – will keep this one in mind

**Reply by nvolker** [reply]
Awesome! Some examples of how [Notion](https://www.notion.so/) handles drag-n-drop media. The value in in-line media for the short-form content world we're transitioning into really excites us 😃 

### Uploading
<img width="576" alt="Screenshot 2022-02-08 at 13 33 16" src="https://user-images.githubusercontent.com/45795784/152979181-6f8e0144-7e29-4a17-8f46-7001fd05e9e5.png">

### Uploaded
<img width="556" alt="Screenshot 2022-02-08 at 13 38 22" src="https://user-images.githubusercontent.com/45795784/152979770-0432588f-7a06-4251-b204-3308c34fd118.png">

**Reply by nvolker** [reply]
Hey @tommoor! I see #859 is soon-to-shipped, https://github.com/outline/outline/pull/3031#issuecomment-1032301142. Will this give users the ability to upload videos as well as have them playable, from within Outline, or will users need to download the media file and play it locally?

**Reply by tommoor** [reply]
It's not included in this first pass so a download is needed. I still think that an in-line player shouldn't be too much of a lift as a follow-on.

**Reply by nvolker** [reply]
Makes sense, thanks for the response. Should I create a separate issue to this for tracking and discussion or is this thread within this discussion suitable?

**Reply by tommoor** [reply]
Feel free to create an issue, it's less likely to get "lost"

**Reply by nvolker** [reply]
Aha, there is only a template for Bug Reports. So I won't be able to create an `enhancement` issue:
<img width="1284" alt="Screenshot 2022-02-15 at 09 49 18" src="https://user-images.githubusercontent.com/45795784/154016348-a03e3c10-af8d-4f3e-9c2c-79be4e25e977.png">

"Feature Request" circles me back to discussions.

**Reply by tommoor** [reply]
Filed here #3130

**Comment by socketpair** [comment]
Reencoding to good format can be done this way:

`ffmpeg -i input.avi -c:v libx264 -crf 23 -c:a aac -movflags faststart output.mp4`

or

`ffmpeg -i input.avi -c:v libx264 -crf 23 -c:a aac -movflags faststart -f ipod output.mp4`

**Reply by socketpair** [reply]
also, note, in typical case we need to encode to different resolutions/qualities, so user can choose one in a player. Also, it's need to be figured out if we need HLS or not. Plus one need to have a button "download" with chosen quality.

**Comment by socketpair** [comment]
Some opensource players:

https://github.com/videojs/video.js (https://videojs.com/advanced?video=disneys-oceans )
https://github.com/jplayer/jPlayer (http://jplayer.org/latest/demo-01-video/)
https://github.com/jPlayer/react-jPlayer
https://github.com/sampotts/plyr (see https://github.com/sampotts/plyr/issues/2408 and https://plyr.io/) 
https://github.com/mediaelement/mediaelement
https://github.com/frankyghost/projekktor

**Comment by socketpair** [comment]
Also, video uploading must be done in a way, that allows re-upload after failure. Typically, file are very big, so uploading may be interrupted, and user should be able to restart uploading from the middle.

I think, everything of this is already implemented in some open source project. It needs to be investigated.

**Comment by Norgat** [comment]
Good day, any changes on this topic?
Or only youtube integration works now?

**Reply by herm4ns** [reply]
https://www.getoutline.com/changelog/embedded-videos
